test/CloneFactory.test.js

```diff
@@ -0,0 +1,65 @@
+const ERC20Mock = artifacts.require('ERC20Mock');
+const ERC20CloneFactory = artifacts.require('ERC20CloneFactory');
+
+const BigNumber = web3.BigNumber;
+
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+contract('CloneFactory', function ([_, owner]) {
+  let master;
+  let clonedOne;
+  let clonedTwo;
+  let factory;
+
+  const _masterSupply = 64;
+  const _clonedSupplyOne = 128;
+  const _clonedSupplyTwo = 512;
+  const _updateValue = 32;
+
+  const initFactory = async function (factoryContract) {
+    const _factory = await factoryContract.new(master.address);
+    factory = {
+      createNew: async function (amount) {
+        const tx = await _factory.createNew(amount);
+        return ERC20Mock.at(tx.logs[0].args.cloneAddress);
+      }
+    };
+  };
+
+  describe('factory initialized with master contract', function () {
+    before(async function () {
+      master = await ERC20Mock.new(owner, _masterSupply);
+      await initFactory(ERC20CloneFactory);
+    });
+
+    describe('cloned instances created', function () {
+      before(async function () {
+        clonedOne = await factory.createNew(_clonedSupplyOne, { from: owner });
+        clonedTwo = await factory.createNew(_clonedSupplyTwo, { from: owner });
+      });
+
+      it('should have correct state after init', async function () {
```

Comment:
It'd be great if we could extract the `ERC20` tests into a behavior, and use it here, though I'm not sure we have an issue to track that.

Reply:
Afraid, I'm not getting you :/

Reply:
Sure! We have some `behavior` test files, in which how a contract behaves is described: for example, [`Ownable.behavior`](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/v2.0.0-rc.2/test/ownership/Ownable.behavior.js). This makes the tests reusable, and we can call them from the [`Ownable test`](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/v2.0.0-rc.2/test/ownership/Ownable.test.js) itself, or from any other test suite. 

It'd be great if we did this for the proxy tests (since we'd be running the same tests, making sure everything works properly), but we have not extracted them into a behavior yet - we can do that in a follow-up PR though, so it's nothing to worry about here :)

Reply:
Oh I see, so I'll leave it for now.